### PR TITLE
Remove support for writing certificates, fqdns etc..in the NVS

### PR DESCRIPTION
**Pull request by npmenard**
Mostly code removal since provisioning is a default builtin in micro-rdk builds

### CODE REVIEW

File: micro-rdk-installer/Cargo.toml [hunk 1]
```diff
@@ -14,23 +14,17 @@ version.workspace = true
 anyhow = { workspace = true }
 clap = { workspace = true, features = ["derive"] }
 crc32fast.workspace = true
-der.workspace = true
```
Comment (acmorrow) [inline micro-rdk-installer/Cargo.toml:17]: Are any of the removals from this file the last ref to a package in the top-level `Cargo.toml`? If so they should probably be removed there too.

**Review by acmorrow — APPROVED** [review]
This all looks fine to me, but I wasn't closely involved with the installer development, so probably good if @gvaradarajan has a look as well.

File: micro-rdk-installer/Cargo.toml (hunk 1 above)
Comment (npmenard) [reply micro-rdk-installer/Cargo.toml:17]: yup, will do that

File: micro-rdk-installer/src/nvs/request.rs [hunk 2]
```diff
@@ -1,37 +1,30 @@
...
-            let rest_of_url = format!("/download/{version}/micro-rdk-server-esp32.bin");
-            release_url.push_str(&rest_of_url)
-        }
-        None => release_url.push_str("latest/download/micro-rdk-server-esp32.bin"),
-    };
+    let release_url = version.map_or_else(
+        || {
+            format!(
+                "{}/{}/{}",
+                RELEASES_BASE_URL, "latest/download/", BINARY_NAME
```
Comment (gvaradarajan) [inline micro-rdk-installer/src/nvs/request.rs:22]: looks like this will result in two slashes after the word "download"? Or am I missing something?

**Review by gvaradarajan — CHANGES_REQUESTED** [review]
LGTM, but found one potential bug?

File: micro-rdk-installer/src/nvs/request.rs (hunk 2 above)
Comment (npmenard) [reply micro-rdk-installer/src/nvs/request.rs:22]: yeah the previous piece of code had it although it was before the download know that I re read it (L30) should I put it there? I don't think the // matter but if it's not needed then we should probably remove it.

**Review by gvaradarajan — APPROVED** [review]
LGTM, pending the removal of now unused packages from the workspace level Cargo.toml

**Pull request merged**